ide/che-core-orion-editor/src/main/java/org/eclipse/che/ide/editor/orion/client/jso/MarkedOverlay.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.editor.orion.client.jso;
+
+/**
+ * This class gives access to the marked.js module from orion editor.
+ *
+ * @author Thomas Mäder
+ */
+
+import com.google.gwt.core.client.JavaScriptObject;
+import java.util.function.Consumer;
+import org.eclipse.che.api.promises.client.Promise;
+import org.eclipse.che.api.promises.client.js.Promises;
+
+public class MarkedOverlay extends JavaScriptObject {
+
+  private static Promise<MarkedOverlay> markedPromise = create();
+
+  protected MarkedOverlay() {}
+
+  public static Promise<MarkedOverlay> getMarkedPromise() {
```

Comment:
'getReady()` sounds better

Reply:
Other classes use this pattern, see org.eclipse.che.ide.editor.orion.client.EditorInitializePromiseHolder.getInitializerPromise()